obspy/io/css/core.py

```diff
@@ -134,11 +136,26 @@ def _read_css(filename, **kwargs):
         else:
             read_fmt = np.dtype(dtype)
             fmt = read_fmt
-        with open(filename, "rb") as fh:
-            fh.seek(offset)
-            data = fh.read(read_fmt.itemsize * npts)
-            data = from_buffer(data, dtype=read_fmt)
-            data = np.require(data, dtype=fmt)
+
+        try:
+            # assumed that the waveform file is not compressed
+            fh = open(dfilename, "rb")
+        except FileNotFoundError:
+            # If does not find the waveform file referenced in the wfdisc,
+            # it will try to open a compressed .gz suffix file instead.
+            try:
+                fh = gzip.open(dfilename + '.gz', "rb")
+            except FileNotFoundError:
+                msg = "Missing waveform file: {}".format(dfilename)
+                warnings.warn(msg, UserWarning)
+                continue
```

Comment:
It looks like previously an exception was raised when an expected file was not found. Now  you are showing a warning and continuing on. I think it would be better to stick to raising an exception, as this should be considered a "broken waveform file" in that case.

```python
try:
    ...
except FileNotFoundError as e:
    try:
        ...
    except FileNotFoundError:
        raise e
```

Simply reraising the first one should be fine. If you want to go the extra mile, you could modify the exception message and add that also no `*.gz` file was found either.

Reply:
I am returning to the previous solution, where the function `_read_css` could raise `FileNotFoundError` exception.
By handling the error exception as warning, I tried to give the `_read_css` function a certain tolerance for data inconsistencies.
However, I understand that the core code policy is more strict.
Now, if the user does not want to disable data from the whole wfdisc file (= a whole day), he must first ensure the consistency of the data.